modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb

```diff
@@ -8,33 +8,31 @@ class MviUsersController < ApplicationController
       skip_before_action :authenticate
       before_action :check_required_headers
 
+      # Allows MVI lookups based on user identity traits (presumably obtained from the upstream
+      # identity provider). Returns a 200 response when the MVI lookup succeeds. Raises an error
+      # otherwise.
       def show
-        user = build_user(build_identity)
+        user_identity = save_requested_identity
         service = MVI::Service.new
-        @mvi_response = service.find_profile(user)
+        @mvi_response = service.find_profile(user_identity)
         raise @mvi_response.error if @mvi_response.error
-        icn_found
+        render json: response_with_icn
       end
 
       private
 
-      def build_user(user_identity)
-        User.new(user_identity.attributes)
-      end
-
-      def icn_found
-        render json:
-          {
-            "id": @mvi_response.profile.icn,
-            "type": 'user-mvi-icn',
-            "data": {
-              "attributes": {
-                "icn": @mvi_response.profile.icn,
-                "first_name": @mvi_response.profile&.given_names&.first,
-                "last_name": @mvi_response.profile&.family_name
-              }
+      def response_with_icn
+        {
+          "id": @mvi_response.profile.icn,
+          "type": 'user-mvi-icn',
+          "data": {
+            "attributes": {
+              "icn": @mvi_response.profile.icn,
+              "first_name": @mvi_response.profile&.given_names&.first,
+              "last_name": @mvi_response.profile&.family_name
             }
           }
+        }
       end
```

Comment:
seems a little weird to me that this lives in the controller, maybe a serializer?